compiler/rustc_ast/src/ast.rs

```diff
@@ -1795,6 +1795,26 @@ pub enum LitIntType {
     Unsuffixed,
 }
 
+impl LitIntType {
+    pub const fn max_literal_value(&self) -> u128 {
+        match self {
+            LitIntType::Signed(IntTy::I8) => i8::MAX as u128,
+            LitIntType::Signed(IntTy::I16) => i16::MAX as u128,
+            LitIntType::Signed(IntTy::I32) => i32::MAX as u128,
+            LitIntType::Signed(IntTy::I64) => i64::MAX as u128,
+            LitIntType::Signed(IntTy::I128) => i128::MAX as u128,
+            LitIntType::Signed(IntTy::Isize) => isize::MAX as u128,
+            LitIntType::Unsigned(UintTy::U8) => u8::MAX as u128,
+            LitIntType::Unsigned(UintTy::U16) => u16::MAX as u128,
+            LitIntType::Unsigned(UintTy::U32) => u32::MAX as u128,
+            LitIntType::Unsigned(UintTy::U64) => u64::MAX as u128,
+            LitIntType::Unsigned(UintTy::U128) => u128::MAX,
+            LitIntType::Unsigned(UintTy::Usize) => usize::MAX as u128,
+            LitIntType::Unsuffixed => u128::MAX,
```

Comment:
```suggestion
            LitIntType::Unsuffixed => i32::MAX,
```

Unsuffixed and unconstrained integer literals should have type `i32`, even if they overflow.

See the old behavior before inlining: https://rust.godbolt.org/z/PMG7ac4d4